geth/proxy/rpc.go

```diff
@@ -0,0 +1,122 @@
+package proxy
+
+import (
+	"errors"
+
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/robertkrimen/otto"
+	"github.com/status-im/status-go/geth/common"
+	"github.com/status-im/status-go/geth/proxy/transactions"
+)
+
+//======================================================================================================
+
+// contains all possible method name which will be
+const (
+	EthSendTransaction = "eth_sendTransaction"
+)
+
+//======================================================================================================
+
+// RPCExecutor defines a function type which takes a giving RPCCall to be executed
+// and returned a response.
+type RPCExecutor func(common.NodeManager, common.RPCCall, otto.FunctionCall) (*otto.Object, error)
+
+// RPCRouter defines a top-level router which sits inbetween calls from other
+// to a external service or a running etherem service.
+type RPCRouter struct {
+	common.NodeManager
+	defaultExecutor  RPCExecutor
+	nonsignExecutors map[string]RPCExecutor
+	signExecutors    map[string]RPCExecutor
+	rpclient         *rpc.Client
+}
+
+// NewRPCRouter returns a new instance of a RPCRouter.
+func NewRPCRouter(manager common.NodeManager) *RPCRouter {
```

Comment:
We can instead just have a `switch` statement that checks the `RPCCall.Method` in the https://github.com/status-im/status-go/blob/45e6688e99acf1f1b1213317efd765eb40fe14f9/geth/proxy/rpc.go#L62, but we still need the `RPCRouter` to activately handle the call to `RPCClient` to return appropriate `rpc.Client` for upstream, else back such into the NodeManager itself.